demo/src/screens/foundationScreens/ColorsScreen.js

```diff
@@ -1,115 +1,313 @@
 import _ from 'lodash';
 import React, {Component} from 'react';
-import {ScrollView, View, Text, StyleSheet, Dimensions, TouchableOpacity} from 'react-native';
-import {Colors, Typography, Toast} from 'react-native-ui-lib';//eslint-disable-line
+import {ScrollView, StyleSheet} from 'react-native';
+import {
+  Colors,
+  Spacings,
+  Assets,
+  View,
+  Text,
+  TouchableOpacity,
+  Icon,
+  Button,
+  TextField,
+  Incubator
+} from 'react-native-ui-lib';
 
+const {Toast} = Incubator;
 
-const {width} = Dimensions.get('window');
-const baseColors = ['grey', 'blue', 'cyan', 'green', 'yellow', 'orange', 'red', 'purple', 'violet'];
+const SYSTEM_COLORS = ['grey', 'violet', 'blue', 'green', 'red', 'yellow', 'orange'];
+const INITIAL_COLOR = Colors.white;
+const BASE_PALETTE = ['1', '5', '10', '20', '30', '40', '50', '60', '70', '80'];
+const TOKENS_CATEGORIES = ['Background', 'Text', 'Icon', 'Outline'];
+const TOKENS_ARRAYS = {};
+TOKENS_CATEGORIES.map(category => (TOKENS_ARRAYS[category] = []));
 
-export default class ColorsScreen extends Component {
+for (const key in Colors) {
+  if (key.startsWith('$background')) {
+    TOKENS_ARRAYS.Background.push(key);
+  } else if (key.startsWith('$text')) {
+    TOKENS_ARRAYS.Text.push(key);
+  } else if (key.startsWith('$icon')) {
+    TOKENS_ARRAYS.Icon.push(key);
+  } else if (key.startsWith('$outline')) {
+    TOKENS_ARRAYS.Outline.push(key);
+  }
+}
 
+class ColorsScreen extends Component {
   constructor(props) {
     super(props);
 
+    this.extraColors = this.props.extraColors;
+
     this.state = {
-      key: '',
-      color: Colors.white,
-      background: Colors.blue40,
-      showToast: false
+      selectedColor: INITIAL_COLOR,
+      searchValue: '',
+      filteredTokens: [],
+      showToast: false,
+      message: undefined
     };
   }
 
-  onPress(key, value) {
-    this.setState({key});
-    if (_.includes(key, '60') || _.includes(key, '70') || _.includes(key, '80') || _.includes(key, 'white')) {
-      this.setState({color: Colors.grey10});
+  scrollViewRef = React.createRef();
+  searchRef = React.createRef();
+
+  scrollToTop = () => {
+    this?.scrollViewRef?.current?.scrollTo({x: 0, y: 0, animated: true});
+  };
+
+  toggleToastVisibility = () => {
+    this.setState({showToast: true});
+  };
+
+  dismissToast = () => {
+    this.setState({showToast: false});
+  };
+
+  onTokenPress = value => {
+    const systemColorName = Colors.getSystemColorByHex(Colors[value].toString(), [
+      ...SYSTEM_COLORS,
+      ...(this.extraColors || []),
```

Comment:
Did you check if it crashes when you spread undefined? If it is, you can set the empty array as the const default to avoid this condition... it will look and read better

Reply:
Yes, I checked for crashes, this is why I used the condition inside the spread here.
Do you mean something like this ?
`const {extraColors=[]} = this.props`

Reply:
Yes